Multiline create procedure script causes UnableToExecuteStatementException

Starting from version `org.jdbi:jdbi3-core:3.31.0` I have `UnableToExecuteStatementException` when running a multiline script to create a procedure in MySQL via JDBI. Versions prior to `3.30.0` (including) don't have this issue. It is still reproduced in the latest `3.41.3` release.

I suspect it might be related to https://github.com/jdbi/jdbi/issues/2021, but not sure. Some other statements, like for example `DELETE FROM tablename WHERE ...;`, successfully pass even when multiline.

JDBI versions: `3.31.0+`
Java version: `11`
MySQL version: `8.0.32`
JDBC Driver: `software.aws.rds:aws-mysql-jdbc:1.1.11`

Test to reproduce:
```gradle
plugins {
    id 'java'
}

repositories {
    mavenCentral()
}

dependencies {
    implementation 'org.jdbi:jdbi3-core:3.31.0'
    runtimeOnly 'software.aws.rds:aws-mysql-jdbc:1.1.11'
    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.1'
    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.8.1'
}

test {
    useJUnitPlatform()
}
```

```java
import org.jdbi.v3.core.Jdbi;
import org.junit.jupiter.api.Test;

public class QweTest {
    private static final Jdbi jdbi = Jdbi.create("jdbc:mysql://localhost:3306/dbname");

    @Test
    public void passesSingleLine() {
        String sqlScript = "CREATE PROCEDURE QWE() " +
                "BEGIN " +
                "END; " +
                "DROP PROCEDURE IF EXISTS QWE;";
        jdbi.withHandle(h -> h.createScript(sqlScript).execute());
    }

    @Test
    public void failsMultiLine() {
        String sqlScript = "CREATE PROCEDURE QWE()\n" +
                "BEGIN\n" +
                "END;\n" +
                "DROP PROCEDURE IF EXISTS QWE;\n";
        jdbi.withHandle(h -> h.createScript(sqlScript).execute());
    }
}
```

Output from `failsMultiLine`:
```
java.sql.BatchUpdateException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'DROP PROCEDURE IF EXISTS QWE' at line 5 [statement:"null", arguments:{positional:{}, named:{}, finder:[]}]
org.jdbi.v3.core.statement.UnableToExecuteStatementException: java.sql.BatchUpdateException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'DROP PROCEDURE IF EXISTS QWE' at line 5 [statement:"null", arguments:{positional:{}, named:{}, finder:[]}]
	at app//org.jdbi.v3.core.statement.Batch.execute(Batch.java:89)
	at app//org.jdbi.v3.core.statement.Script.execute(Script.java:43)
	at app//QweTest.lambda$failsMultiLine$1(QweTest.java:24)
	at app//org.jdbi.v3.core.Jdbi.withHandle(Jdbi.java:357)
	at app//QweTest.failsMultiLine(QweTest.java:24)
	at java.base@11.0.18/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.18/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.18/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.18/java.lang.reflect.Method.invoke(Method.java:566)
	at app//org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at app//org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at app//org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at app//org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at app//org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at app//org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base@11.0.18/java.util.ArrayList.forEach(ArrayList.java:1541)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at app//org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base@11.0.18/java.util.ArrayList.forEach(ArrayList.java:1541)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at app//org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at app//org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at app//org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at java.base@11.0.18/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.18/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.18/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.18/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: java.sql.BatchUpdateException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'DROP PROCEDURE IF EXISTS QWE' at line 5
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.util.Util.handleNewInstance(Util.java:193)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.util.Util.getInstance(Util.java:168)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.util.Util.getInstance(Util.java:175)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.exceptions.SQLError.createBatchUpdateException(SQLError.java:224)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.StatementImpl.executeBatchInternal(StatementImpl.java:893)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.StatementImpl.executeBatch(StatementImpl.java:797)
	at org.jdbi.v3.core.statement.SqlLoggerUtil.wrap(SqlLoggerUtil.java:31)
	at org.jdbi.v3.core.statement.Batch.execute(Batch.java:87)
	... 89 more
Caused by: java.sql.SQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'DROP PROCEDURE IF EXISTS QWE' at line 5
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:120)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.StatementImpl.executeUpdateInternal(StatementImpl.java:1339)
	at software.aws.rds.jdbc.mysql.shading.com.mysql.cj.jdbc.StatementImpl.executeBatchInternal(StatementImpl.java:859)
	... 92 more
```

--- a/core/src/main/antlr4/org/jdbi/v3/core/internal/lexer/SqlScriptLexer.g4
+++ b/core/src/main/antlr4/org/jdbi/v3/core/internal/lexer/SqlScriptLexer.g4
@@ -48,7 +48,7 @@ BLOCK_BEGIN
     ;
 
 BLOCK_END
-    : 'END' WHITESPACE+
+    : 'END' WHITESPACE*
     ;
 
 fragment ESCAPE_SEQUENCE

--- a/core/src/main/java/org/jdbi/v3/core/internal/SqlScriptParser.java
+++ b/core/src/main/java/org/jdbi/v3/core/internal/SqlScriptParser.java
@@ -13,6 +13,9 @@
  */
 package org.jdbi.v3.core.internal;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import org.antlr.v4.runtime.CharStream;
 import org.antlr.v4.runtime.Token;
 import org.jdbi.v3.core.internal.lexer.SqlScriptLexer;
@@ -36,10 +39,10 @@ import static org.jdbi.v3.core.internal.lexer.SqlScriptLexer.SEMICOLON;
  */
 public class SqlScriptParser {
 
-    private final TokenHandler semicolonHandler;
+    private final TokenHandler tokenHandler;
 
-    public SqlScriptParser(TokenHandler semicolonHandler) {
-        this.semicolonHandler = semicolonHandler;
+    public SqlScriptParser(TokenHandler tokenHandler) {
+        this.tokenHandler = tokenHandler;
     }
 
     public String parse(CharStream charStream) {
@@ -47,38 +50,31 @@ public class SqlScriptParser {
         SqlScriptLexer lexer = new SqlScriptLexer(charStream);
         lexer.addErrorListener(new ErrorListener());
         boolean endOfFile = false;
-        int blockLevel = 0;
         while (!endOfFile) {
             Token t = lexer.nextToken();
             switch (t.getType()) {
+                // EOF ends parsing
                 case Token.EOF:
                     endOfFile = true;
                     break;
-                case SEMICOLON:
-                    if (blockLevel == 0) {
-                        semicolonHandler.handle(t, sb);
-                    } else {
-                        // preserve semicolons within begin/end block
-                        sb.append(t.getText());
-                    }
-                    break;
-                case BLOCK_BEGIN:
-                case BLOCK_END:
-                    blockLevel += BLOCK_BEGIN == t.getType() ? +1 : -1;
-                    sb.append(t.getText());
-                    break;
+                // Strip comments out
                 case COMMENT:
                 case MULTI_LINE_COMMENT:
                     break;
+                // collapse newlines
                 case NEWLINES:
                     if (sb.length() > 0) {
                         sb.append('\n');
                     }
                     break;
+                // everything else is done by the handler
+                case SEMICOLON:
+                case BLOCK_BEGIN:
+                case BLOCK_END:
                 case QUOTED_TEXT:
                 case LITERAL:
                 case OTHER:
-                    sb.append(t.getText());
+                    tokenHandler.handle(t, sb);
                     break;
                 default:
                     throw new IllegalArgumentException("Unrecognizable token " + t);
@@ -89,5 +85,57 @@ public class SqlScriptParser {
 
     public interface TokenHandler {
         void handle(Token t, StringBuilder sb);
+    }
+
+    public static final class ScriptTokenHandler implements TokenHandler {
+
+        private final List<String> statements = new ArrayList<>();
+        private Token lastToken = null;
+
+        private int blockLevel = 0;
+
+        @Override
+        public void handle(Token t, StringBuilder sb) {
+            switch (t.getType()) {
+                case BLOCK_BEGIN:
+                    blockLevel++;
+                    sb.append(t.getText());
+                    break;
+                case BLOCK_END:
+                    if (blockLevel == 0) {
+                        throw new IllegalStateException("Found END in SQL script without BEGIN!");
+                    }
+
+                    blockLevel--;
+                    sb.append(t.getText());
+                    break;
+                case SEMICOLON:
+                    if (blockLevel == 0) {
+                        if (lastToken != null && lastToken.getType() == BLOCK_END) {
+                            sb.append(t.getText());
+                        }
+                        addStatement(sb.toString());
+                        sb.setLength(0);
+                    } else {
+                        sb.append(t.getText());
+                    }
+                    break;
+                default:
+                    sb.append(t.getText());
+            }
+            lastToken = t;
+        }
+
+        public List<String> getStatements() {
+            return statements;
+        }
+
+        public List<String> addStatement(String statement) {
+            String trimmedStatement = statement.trim();
+            if (!trimmedStatement.isEmpty()) {
+                statements.add(trimmedStatement);
+            }
+            return statements;
+        }
     }
 }

--- a/core/src/main/java/org/jdbi/v3/core/locator/ClasspathSqlLocator.java
+++ b/core/src/main/java/org/jdbi/v3/core/locator/ClasspathSqlLocator.java
@@ -115,10 +115,8 @@ public final class ClasspathSqlLocator {
      * @return A new ClasspathSqlLocator that returns SQL with comments removed.
      */
     public static ClasspathSqlLocator removingComments() {
-        final SqlScriptParser commentStripper =
-                new SqlScriptParser((t, sb) -> sb.append(t.getText()));
-        return new ClasspathSqlLocator(
-                r -> commentStripper.parse(CharStreams.fromStream(r)));
+        final SqlScriptParser commentStripper = new SqlScriptParser((t, sb) -> sb.append(t.getText()));
+        return new ClasspathSqlLocator(r -> commentStripper.parse(CharStreams.fromStream(r)));
     }
 
     /**

--- a/core/src/main/java/org/jdbi/v3/core/statement/Script.java
+++ b/core/src/main/java/org/jdbi/v3/core/statement/Script.java
@@ -13,12 +13,12 @@
  */
 package org.jdbi.v3.core.statement;
 
-import java.util.ArrayList;
 import java.util.List;
 
 import org.antlr.v4.runtime.CharStreams;
 import org.jdbi.v3.core.Handle;
 import org.jdbi.v3.core.internal.SqlScriptParser;
+import org.jdbi.v3.core.internal.SqlScriptParser.ScriptTokenHandler;
 
 /**
  * Represents a number of SQL statements delimited by semicolon which will be executed in order in a batch statement.
@@ -68,21 +68,10 @@ public class Script extends SqlStatement<Script> {
     }
 
     private List<String> splitToStatements(String script) {
-        final List<String> statements = new ArrayList<>();
-        String lastStatement = new SqlScriptParser((t, sb) -> {
-            addStatement(sb.toString(), statements);
-            sb.setLength(0);
-        }).parse(CharStreams.fromString(script));
-        addStatement(lastStatement, statements);
+        ScriptTokenHandler scriptTokenHandler = new ScriptTokenHandler();
+        String lastStatement = new SqlScriptParser(scriptTokenHandler)
+            .parse(CharStreams.fromString(script));
 
-        return statements;
-    }
-
-    private void addStatement(String statement, List<String> statements) {
-        String trimmedStatement = statement.trim();
-        if (trimmedStatement.isEmpty()) {
-            return;
-        }
-        statements.add(trimmedStatement);
+        return scriptTokenHandler.addStatement(lastStatement);
     }
 }

--- a/mysql/src/test/java/org/jdbi/v3/mysql/TestMySQL.java
+++ b/mysql/src/test/java/org/jdbi/v3/mysql/TestMySQL.java
@@ -45,7 +45,7 @@ public class TestMySQL {
 
     @RegisterExtension
     JdbiExtension extension = JdbiTestcontainersExtension.instance(dbContainer)
-            .withPlugin(new SqlObjectPlugin());
+        .withPlugin(new SqlObjectPlugin());
 
     @Test
     void testIssue2402() {
@@ -76,13 +76,13 @@ public class TestMySQL {
         List<Contact> getAllContacts();
 
         @SqlQuery("SELECT contact_id "
-                + " FROM contacts "
-                + " WHERE RIGHT(etag, 1) = RIGHT(:etag, 1)")
+            + " FROM contacts "
+            + " WHERE RIGHT(etag, 1) = RIGHT(:etag, 1)")
         List<String> testOne(@Bind("etag") String etag);
 
         @SqlQuery("SELECT contact_id"
-                + " FROM contacts "
-                + " WHERE etag LIKE \\'%<etagPattern>\\'")
+            + " FROM contacts "
+            + " WHERE etag LIKE \\'%<etagPattern>\\'")
         List<String> testTwo(@Define("etagPattern") String etag);
     }
 
@@ -103,5 +103,28 @@ public class TestMySQL {
         public String etag() {
             return etag;
         }
+    }
+
+
+    @Test
+    public void testIssue2535PassesSingleLine() {
+        String sqlScript = "CREATE PROCEDURE QWE() "
+            + "BEGIN "
+            + "END; "
+            + "DROP PROCEDURE IF EXISTS QWE;";
+        int[] result = extension.getJdbi().withHandle(h -> h.createScript(sqlScript).execute());
+
+        assertThat(result).isEqualTo(new int[] {0, 0 });
+    }
+
+    @Test
+    public void testIssue2535FailsMultiLine() {
+        String sqlScript = "CREATE PROCEDURE QWE()\n"
+            + "BEGIN\n"
+            + "END;\n"
+            + "DROP PROCEDURE IF EXISTS QWE;\n";
+        int[] result = extension.getJdbi().withHandle(h -> h.createScript(sqlScript).execute());
+
+        assertThat(result).isEqualTo(new int[] {0, 0});
     }
 }

